### PR TITLE
matrix: make classes and methods package private in reference solution. Resolved the issue

### PR DESCRIPTION
**Pull request by pratikpalashikar**
…ss modifier

I have made the class as package private and removed the public access modifier from the public methods as well. Removed the extra spaces as well to preserve the indentation.  PR #1083

Edited the class Hamming to be package private, retaining the private method as it is. PR #1049 

---

Reviewer Resources:

[Track Policies](https://github.com/exercism/xjava/blob/master/POLICIES.md#event-checklist)

### CODE REVIEW

File: exercises/matrix/.meta/src/reference/java/Matrix.java [hunk 1]
```diff
@@ -1,11 +1,11 @@
 import java.util.regex.Pattern;
 
-public class Matrix {
+ class Matrix {
```
Comment (FridaTveit) [inline exercises/matrix/.meta/src/reference/java/Matrix.java]: I think all the lines you've changed you've indented one space too much? Try to keep the indentation the same as it was :)

File: exercises/matrix/.meta/src/reference/java/Matrix.java [hunk 2]
```diff
@@ -1,11 +1,11 @@
...
     private int[][] matrix;
     private static Pattern spacePattern = Pattern.compile(" ");
     private static Pattern newlinePattern = Pattern.compile("\\n");
 
-    public Matrix(String matrixAsString) {
+     public Matrix(String matrixAsString) {
```
Comment (FridaTveit) [inline exercises/matrix/.meta/src/reference/java/Matrix.java]: This can also be package private :)

Comment (pratikpalashikar) [reply exercises/matrix/.meta/src/reference/java/Matrix.java]: @FridaTveit thank you for suggestions! I will take care of it.

File: exercises/matrix/.meta/src/reference/java/Matrix.java (hunk 1 above)
Comment (pratikpalashikar) [reply exercises/matrix/.meta/src/reference/java/Matrix.java]: Sure !

**Comment by pratikpalashikar** [comment]
@FridaTveit  I have integrated the changes, please let me know if any more changes are needed!

**Comment by pratikpalashikar** [comment]
Hi @FridaTveit  I have added the changes for the PR #1049  also. Sorry for including the different PR changes in this one. I will take care of it next time.

File: exercises/hamming/.meta/src/reference/java/Hamming.java [hunk 3]
```diff
@@ -1,15 +1,15 @@
...
         hammingDistance = computeHammingDistance(leftStrand, rightStrand);
     }
 
-    public int getHammingDistance() {
+    int getHammingDistance() {
         return hammingDistance;
     }
 
-    private int computeHammingDistance(String leftStrand, String rightStrand) {
+    int computeHammingDistance(String leftStrand, String rightStrand) {
```
Comment (FridaTveit) [inline exercises/hamming/.meta/src/reference/java/Hamming.java]: This (and the next method) should remain private. We want to be as restrictive as possible with access modifiers and private is more restrictive than package private, see https://docs.oracle.com/javase/tutorial/java/javaOO/accesscontrol.html for an explanation :)

Comment (pratikpalashikar) [reply exercises/hamming/.meta/src/reference/java/Hamming.java]: @FridaTveit  Thank you for your suggestion. I am just starting to contribute so its really helpful. And I thought that we have to make all the things package private even the private methods. Thank you for the explanation!

**Comment by pratikpalashikar** [comment]
@FridaTveit  I have committed the changes related to the hamming class and also updated the description.

**Pull request merged**